### ISSUE TITLE
Let Examples Choose GPU

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu-rs#703]
We already have ways to choose backend, for devs like me with many gpus in one system, it would be helpful to have a way in the framework to override the gpu based on env. Obviously `enumerate_adapters` isn't available on wasm, so this would have to be ignored.

**Comment by kvark** [comment]
How do you see this working? `WGPU_ADAPTER="Nvidia"` matched for the name string?

**Comment by cwfitzgerald** [comment]
Yeah, that's what I do in my code, I just lowercase both the desired name and the adapter name, then do `adapter_name.contains(desired)`, and if that returns true, I select that adapter.
